### PR TITLE
Send OC version to appstore so it can reply with the right links

### LINKED ISSUE #17101: AppStore does not return applications for 8.1
Request sent:

```
GET /v1/content/data?version=8x1x0x7&filter=all&categories=920&sortmode=new&page=0&pagesize=100&approved=all HTTP/1.1
Host: api.owncloud.com
User-Agent: ownCloud Server Crawler
```

Response:
```
HTTP/1.0 200 OK
Date: Tue, 23 Jun 2015 11:33:05 GMT
Server: Apache/2.4.7 (Ubuntu)
X-Powered-By: PHP/5.5.9-1ubuntu4.9
X-Frame-Options: SAMEORIGIN
Set-Cookie: PHPSESSID=h6o149gj2m2bo7kcie3hkskh72; path=/; HttpOnly
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Vary: Accept-Encoding
Content-Length: 200
Connection: close
Content-Type: application/xml

<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>100</statuscode>
  <message></message>
  <totalitems>0</totalitems>
  <itemsperpage>100</itemsperpage>
 </meta>
 <data/>
</ocs>
```

Why does it not return applications such as [News](https://apps.owncloud.com/content/show.php/News?content=168040) which is set to 8.1 and also approved?

@karlitschek Any idea?

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #17101 

@DeepDiver1975 @LukasReschke @karlitschek

### CODE REVIEW

File: lib/private/ocsclient.php [hunk 1]
```diff
@@ -121,6 +121,9 @@ public function getCategories() {
 				$this->getAppStoreUrl() . '/content/categories',
 				[
 					'timeout' => 5,
+					'query' => [
+						'version' => implode('x', \OC_Util::getVersion()),
```
Comment (DeepDiver1975) [inline lib/private/ocsclient.php]: can we inject the version?

Comment (nickvergessen) [inline lib/private/ocsclient.php]: everywhere?

**Comment by DeepDiver1975** [comment]
@nickvergessen :boom: :tada: 

````
Test Result (10 failures / +6)

    OCSClientTest.testGetCategoriesExceptionClient
    OCSClientTest.testGetCategoriesParseError
    OCSClientTest.testGetCategoriesSuccessful
    OCSClientTest.testGetApplicationExceptionClient
    OCSClientTest.testGetApplicationParseError
    OCSClientTest.testGetApplicationSuccessful
    OCSClientTest.testGetApplicationEmptyXml
    OCSClientTest.testGetApplicationDownloadExceptionClient
    OCSClientTest.testGetApplicationDownloadParseError
    OCSClientTest.testGetApplicationDownloadUrlSuccessful

**Comment by nickvergessen** [comment]
@DeepDiver1975 injected the variable everywhere

**Comment by karlitschek** [comment]
:+1: Please also backport to stable 6,7 and 8
Thanks

**Comment by DeepDiver1975** [comment]
> Please also backport to stable 6,7 and 8
> Thanks

@nickvergessen can you please take care? THX

**Comment by nickvergessen** [comment]
> @nickvergessen can you please take care? THX

Of course, but I'd prefer to not rebase multiple times, which is why a merged master would be awesome to have :)

**Comment by LukasReschke** [comment]
:+1:

**Comment by LukasReschke** [comment]
Discovered a small problem with it afterwards (caching *sigh*), please go to https://github.com/owncloud/core/pull/17231 :speak_no_evil:

**Pull request merged**